### PR TITLE
NH-3726: Added escape character support in SqlMethods.Like

### PR DESCRIPTION
**Pull request by gliljas**
Did not yet follow my own suggestion to fix StartsWIth, Contains and EndWith, since that could be considered a breaking change. But it is a bug, IMO.

### CODE REVIEW

**Comment by hazzik** [comment]
>All checks have failed

**Comment by gliljas** [comment]
Hmm...

I'm on a shaky rural connection, but will try to fix ASAP

**Comment by gliljas** [comment]
Fixed and squashed

File: src/NHibernate.Test/Linq/FunctionTests.cs [hunk 1]
```diff
@@ -23,6 +23,38 @@ where NHibernate.Linq.SqlMethods.Like(e.FirstName, "Ma%et")
...
+		{
+			using (var tx = session.BeginTransaction())
+			{
+				var employeeName = "Mar%aret";
+				var escapeChar = '\\';
+				var employeeNameEscaped = employeeName.Replace("%", escapeChar + "%");
+
+				//This entity will be flushed to the db, but rolled back when the test completes
+
+				session.Save(new Employee { FirstName = employeeName, LastName = "" });
```
Comment (hazzik) [inline src/NHibernate.Test/Linq/FunctionTests.cs]: Oracle does not like `""` as `LastName` because in oracle `""` is `NULL`

**Comment by oskarb** [comment]
I suspect it's the choice of escape character that is causing the fail in MySql. I will test.

**Comment by oskarb** [comment]
I've fixed the issue on MySql, but decided to also rebase to get rid of unrelated formatting changes. Merged separately.

**Pull request closed without merging**